File: cells.py
x = [1, 2, 50, 3213, 12, 2, 1]


def cell(a, c):
    cell_width = len(str(max(a)))
    row_art = '-' * cell_width
    padding = '{:>' + str(cell_width) + '}|'

    print("+", end="")
    column_count = int()
    for _ in a:
        if column_count == c:
            break
        column_count += 1
        print(row_art + "+", end="")

    print("\n|", end="")
    column_count = 0
    for i in a:
        if column_count == c:
            print("\n+", end="")
            for _ in range(c):
                print(row_art + "+", end="")
            print("\n|", end="")
            column_count = 0
        column_count += 1
        print(padding.format(i), end="")
    print("\n+", end="")
    if len(a) % c == 0 and c != 1:
        for _ in range(c):
            print(row_art + "+", end="")
    elif c == 1:
        print(row_art + "+")
    else:
        for _ in range(len(a) % c):
            print(row_art + "+", end="")


cell(x, 4)

File: cells.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

/*
   +---+---+
   | 10| 20|
   +---+---+
   |400|
   +---+
*/

func main() {
	x := []int{1, 10, 999, 9999999, 7, 6, 15, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1}
	table(x, 2)
}

func cellWidth(s []int) int {
	var cellWidth int
	for _, i := range s {
		if i > cellWidth {
			cellWidth = i
		}
	}

	return len(fmt.Sprintf("%d", cellWidth))
}

func table(s []int, c int) {
	cw := cellWidth(s)
	rowArt := strings.Repeat("-", cw)
	fmt.Print("+")
	var columnCount int
	for range s {
		if columnCount == c {
			break
		}
		columnCount++
		fmt.Printf("%s+", rowArt)
	}
	columnCount = 0
	fmt.Print("\n|")
	for _, i := range s {
		if columnCount == c {
			columnCount = 0
			fmt.Print("\n+")
			for j := 0; j < c; j++ {
				fmt.Printf("%s+", rowArt)
			}
			fmt.Print("\n|")
		}
		columnCount++
		fmt.Printf("%"+strconv.Itoa(cw)+"d|", i)
	}
	fmt.Printf("\n+")

	if len(s)%c == 0 && c != 1 {
		for range s {
			fmt.Printf("%s+", rowArt)
		}
	} else if c == 1 {
		fmt.Printf("%s+", rowArt)
	} else {
		for i := 0; i < len(s)%c; i++ {
			fmt.Printf("%s+", rowArt)
		}
	}
	fmt.Print("\n")
}
